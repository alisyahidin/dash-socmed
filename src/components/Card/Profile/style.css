.profile-avatar {
  margin-bottom: 0.5rem;
}

.profile-username {
  color: #888;
}