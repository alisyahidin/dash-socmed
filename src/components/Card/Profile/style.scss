.profile-card {
  &__avatar {
    margin-bottom: 0.5rem;
  }
  &__username {
    color: #888;
  }
}