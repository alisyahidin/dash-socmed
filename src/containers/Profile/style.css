.profile-tabs {
  width: 100%;
  border-bottom: 1px solid #dee2e6 !important;
}

.profile-tabs > a {
  color: rgba(0, 0, 0, 0.5);
}

.tab-content {
  width: 100%;
}

.nav-pills > .nav-link.active {
  color: #FFF !important;
}