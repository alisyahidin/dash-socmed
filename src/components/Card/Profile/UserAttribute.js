import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-bootstrap'
import './style.scss'

const UserAttribute = ({label, value}) => {
  return (
    <div className="d-flex w-100">
      <Form.Label column sm="4">
        <b>{label}</b>
      </Form.Label>
      <Form.Label column>
        {value}
      </Form.Label>
    </div>
  )
}

UserAttribute.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
}

UserAttribute.defaultProps = {
  label: 'label',
  value: 'value',
}

export default UserAttribute