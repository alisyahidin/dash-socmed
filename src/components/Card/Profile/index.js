import React from 'react'
import PropTypes from 'prop-types'
import Avatar from 'react-avatar'
import Card from '../'
import './style.css'

const Profile = ({name, email}) => (
  <Card align="center">
    <Avatar round size="70px" name={name} className="profile-avatar" />
    <h4>{name}</h4>
    <p className="profile-email">{email}</p>
  </Card>
)

Profile.propTypes = {
  name: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
}

Profile.defaultProps = {
  name: 'Guest',
  email: 'quest@email.com',
}

export default Profile