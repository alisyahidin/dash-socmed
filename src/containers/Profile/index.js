import React, { Component } from 'react'
import {
  Container,
  Row,
  Col,
  Tabs,
  Tab,
  Nav,
  Image
} from 'react-bootstrap'

import withNavbar from '../../hoc/withNavbar'
import ProfileCard from '../../components/Card/Profile/'
import PostCard from '../../components/Card/Post/'
import Card from '../../components/Card/'
import './style.css'

import img from '../../assets/images/placeholder.png'

const posts = [
  {
    userId: 1,
    title: 'Learn NextJS',
    body: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus accusantium, eos! Quibusdam molestiae, voluptatem veritatis odit ducimus ratione delectus dolorum placeat et, ea repellendus sit obcaecati sint hic quis qui!'
  },
  {
    userId: 1,
    title: 'Reactive Programming is Amazing',
    body: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus accusantium, eos! Quibusdam molestiae, voluptatem veritatis odit ducimus ratione delectus dolorum placeat et, ea repellendus sit obcaecati sint hic quis qui!'
  },
]

class Profile extends Component {
  render() {
    return (
      <Container>
        <Row>
          <Col md={4}>
            <ProfileCard name="Ali Syahidin" username="alisyahidin" />
          </Col>
          <Col md={8}>
            <Card align="left" className="mb-sm-1">
              <Tabs defaultActiveKey="posts" className="profile-tabs m-sm-0">
                <Tab eventKey="posts" title="Posts">
                  {posts.map((data, index) => (
                    <PostCard author={{ name: 'Ali Syahidin', username: 'alisyahidin'}} key={index} {...data} />
                  ))}
                </Tab>
                <Tab eventKey="photos" title="Photos">
                  <Tab.Container id="albums" defaultActiveKey="first">
                    <Row className="m-sm-0 mt-sm-4">
                      <Col sm={3}>
                        <Nav variant="pills" className="flex-column">
                          <Nav.Item>
                            <Nav.Link eventKey="first">Album 1</Nav.Link>
                          </Nav.Item>
                          <Nav.Item>
                            <Nav.Link eventKey="second">Album 2</Nav.Link>
                          </Nav.Item>
                        </Nav>
                      </Col>
                      <Col sm={9}>
                        <Tab.Content>
                          <Tab.Pane eventKey="first">
                            <Image className="w-25" src={img} thumbnail />
                            <Image className="w-25" src={img} thumbnail />
                            <Image className="w-25" src={img} thumbnail />
                            <Image className="w-25" src={img} thumbnail />
                            <Image className="w-25" src={img} thumbnail />
                            <Image className="w-25" src={img} thumbnail />
                            <Image className="w-25" src={img} thumbnail />
                          </Tab.Pane>
                          <Tab.Pane eventKey="second">
                            <Image className="w-25" src={img} thumbnail />
                            <Image className="w-25" src={img} thumbnail />
                            <Image className="w-25" src={img} thumbnail />
                          </Tab.Pane>
                        </Tab.Content>
                      </Col>
                    </Row>
                  </Tab.Container>
                </Tab>
              </Tabs>
            </Card>
          </Col>
        </Row>
      </Container>
    )
  }
}

export default withNavbar(Profile)
