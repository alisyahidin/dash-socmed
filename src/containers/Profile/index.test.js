import React from 'react'
import { shallow } from 'enzyme'
import Profile from './index'

it('render <Profile /> correctly', () => {
  const component = shallow(<Profile />)

  expect(component).toMatchSnapshot()
})