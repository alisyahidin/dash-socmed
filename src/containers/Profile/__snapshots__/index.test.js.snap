// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Profile /> correctly 1`] = `
<Fragment>
  <withRouter(NavigationBar) />
  <div
    className="main-content"
  >
    <Profile />
  </div>
</Fragment>
`;
